core/src/test/java/com/linecorp/armeria/client/ConnectionPoolCollectingMetricTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.net.InetSocketAddress;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.metric.MoreMeters;
+import com.linecorp.armeria.common.metric.PrometheusMeterRegistries;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.netty.util.AttributeMap;
+import io.netty.util.DefaultAttributeMap;
+
+@SuppressWarnings("checkstyle:LineLength")
+public class ConnectionPoolCollectingMetricTest {
+    private ConnectionPoolListener connectionPoolListener;
+    final MeterRegistry registry = PrometheusMeterRegistries.newRegistry();
+
+    @BeforeEach
+    void setUp() {
+        connectionPoolListener = ConnectionPoolListener.metricCollecting(registry);
```

Comment:
It would be better to create a new `MeterRegistry` before starting a new test since it can be dirty because of the previous test in case.

Reply:
Updated. Thanks